File: src/main/java/com/interviewplannerapp/service/AccountService.java
package com.interviewplannerapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.interviewplannerapp.domain.Account;
import com.interviewplannerapp.dto.AccountDTO;
import com.interviewplannerapp.dto.AccountSearchDTO;
import com.interviewplannerapp.dto.AccountPageDTO;
import com.interviewplannerapp.dto.AccountConvertCriteriaDTO;
import com.interviewplannerapp.service.GenericService;
import com.interviewplannerapp.dto.common.RequestDTO;
import com.interviewplannerapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AccountService extends GenericService<Account, Integer> {

	List<Account> findAll();

	ResultDTO addAccount(AccountDTO accountDTO, RequestDTO requestDTO);

	ResultDTO updateAccount(AccountDTO accountDTO, RequestDTO requestDTO);

    Page<Account> getAllAccounts(Pageable pageable);

    Page<Account> getAllAccounts(Specification<Account> spec, Pageable pageable);

	ResponseEntity<AccountPageDTO> getAccounts(AccountSearchDTO accountSearchDTO);
	
	List<AccountDTO> convertAccountsToAccountDTOs(List<Account> accounts, AccountConvertCriteriaDTO convertCriteria);

	AccountDTO getAccountDTOById(Integer accountId);







}

File: interviewplannerapp_ui/src/services/AccountService.js
import http from "../http-common"; 

class AccountService {
  getAllAccounts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/account/accounts`, searchDTO);
  }

  get(accountId) {
    return this.getRequest(`/account/${accountId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/account?field=${matchData}`, null);
  }

  addAccount(data) {
    return http.post("/account/addAccount", data);
  }

  update(data) {
  	return http.post("/account/updateAccount", data);
  }
  
  uploadImage(data,accountId) {
  	return http.postForm("/account/uploadImage/"+accountId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AccountService();
